Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs

```diff
@@ -498,41 +511,82 @@ public ActionResult StatsChart(int contestId)
                     .Select(ContestProblemListViewModel.FromProblem),
             };
 
+            var participantsToSkip = 0;
+            var participantsToTake = int.MaxValue;
+
+            if (page.HasValue)
+            {
+                participantsToSkip = (page.Value - 1) *
+                                     (official
+                                         ? OfficialResultsPageSize
+                                         : NotOfficialResultsPageSize);
+
+                participantsToTake = official
+                    ? OfficialResultsPageSize
+                    : NotOfficialResultsPageSize;
+            }
+
             var participants = this.participantsData
                 .GetAllByContestAndIsOfficial(contest.Id, official)
-                .AsNoTracking();
+                .OrderByDescending(p => p.TotalScoreSnapshot)
+                .ThenBy(p => p.TotalScoreSnapshotModifiedOn)
+                .Skip(participantsToSkip)
+                .Take(participantsToTake)
+                .ToList();
```

Comment:
Try not to do `.ToList` here, as this will get all the properties from db. Also you will make extra calls from lazy loading of User and Scores, when you construct the `ParticipantResultViewModel` later on in the code.
You should `.ToList` after the projection to the resulting model.
This will probably fix the exception you mentioned that made you move logic inside the controller.

Reply:
If we remove .ToList() the returned data is not ordered. This is likely because when we remove .ToList() the query remains in "deferred state", next manipulations of the query is likely to interfere with the ordering

Reply:
This seems strange. I would like to check it out with you, when you have the chance. Maybe we can figure out what is happening, or/and learn something new.

Reply:
Fixed problem with N+1 queries and getting all the properties.